gips/0054-timeline-aggregation-protocol.md

```diff
@@ -0,0 +1,270 @@
+| - |
+| <b>Fig. 5: (Mandatory) receipt checks performed by the Indexer when preparing RAV request</b>|
+
+Please refer to [TAP Manager documentation](https://docs.rs/tap_core/0.1.0/tap_core/tap_manager/struct.Manager.html) for more details on this topic.
+
+
+
+### Gateway Receipt Aggregator
+
+Gateway Receipt Aggregator is a A JSON-RPC service that lets clients request an aggregate receipt (RAV) based on a list of individual receipts. High-level specification can be found [TAP Aggregator README](https://docs.rs/crate/tap_aggregator/0.1.0).
+
+Detailed documentation can be found in [TAP Aggregator docs.rs](https://docs.rs/tap_core/0.1.0/tap_core/).
+
+## Smart Contracts (Tom/Bryan | Status: OK)
+
+TAPVerifier - A contract for verifying receipt aggregation vouchers.
+
+Collateral - This contract allows `senders` to deposit collateral for specific `receivers`, which can later be redeemed using Receipt Aggregate Vouchers (`RAV`) signed by an authorized `signer`. `Senders` can deposit collateral for `receivers`, authorize `signers` to create signed `RAVs`, and withdraw collateral after a set `thawingPeriod` number of seconds. `Receivers` can redeem signed `RAVs` to claim collateral.
+This contract uses the `TAPVerifier` contract for recovering signer addresses from `RAVs`.
+
+AllocationIDTracker - This contract tracks the allocation IDs of the RAVs that have been submitted to ensure that each allocation ID is only used once. It is external to the collateral contract to allow for updating the collateral contract without losing the list of used allocation IDs.
+This contract is intended to be used with the `Collateral` contract.
+
+Implementation along with detailed API documentation of TAP contracts can be found [here](https://github.com/semiotic-ai/timeline-aggregation-protocol-contracts).
+
+# Backwards Compatibility
+
+## Gateway:
+Indexers are currently expected to report their software version (via the `/version` path of indexer-service). During the transition period, the gateway will use this version to determine if the indexer can be expected to support TAP. If so, the gateway will only send receipts to that indexer using TAP. Otherwise, the current Scalar receipt format will be used. This will ensure that any query sent to an indexer from a gateway will only be associated with one receipt, and that receipt will only be associated with one of the two Scalar protocol versions.
+
+## Indexers:
+We assume no backwards compatibility for the Indexer Service, i.e. once the Indexers will upgrate their stack, then they will switch to the TAP Scalar payments.
+Hence the protocol will have to make advertise that this is a breaking change and Indexers will have to redeem your receipts and close all the allocations before this upgrade.
+
+## Smart contacts:
+Similarly to Indexer Service, update to new stack will automatically trigger switching to new Collateralization and TAP Verifier smart contracts.
```

Comment:
This should be seamless from a user perspective (i.e. this is covered by the Indexer stack changes)?

Reply:
correct, this will be "invisible" from the perspective of running an Indexer